src/main/java/appeng/thirdparty/codechicken/lib/model/pipeline/transformers/QuadReInterpolator.java

```diff
@@ -0,0 +1,62 @@
+package appeng.thirdparty.codechicken.lib.model.pipeline.transformers;
+
+import appeng.thirdparty.codechicken.lib.math.InterpHelper;
+import appeng.thirdparty.codechicken.lib.model.CachedFormat;
+import appeng.thirdparty.codechicken.lib.model.Quad;
+import appeng.thirdparty.codechicken.lib.model.Quad.Vertex;
+import appeng.thirdparty.codechicken.lib.model.pipeline.IPipelineElementFactory;
+import appeng.thirdparty.codechicken.lib.model.pipeline.QuadTransformer;
+
+/**
+ * This transformer Re-Interpolates the Color, UV's and LightMaps.
+ * Use this after all transformations that translate vertices in the pipeline.
+ *
+ * This Transformation can only be used in the BakedPipeline.
+ *
+ * Created by covers1624 on 30/07/18.
+ */
+public class QuadReInterpolator extends QuadTransformer {
+
+    public static final IPipelineElementFactory<QuadReInterpolator> FACTORY = QuadReInterpolator::new;
+
+    private Quad interpCache = new Quad();
+    private InterpHelper interpHelper = new InterpHelper();
+
+    QuadReInterpolator() {
+        super();
+    }
+
+    @Override
+    public void reset(CachedFormat format) {
+        super.reset(format);
+        interpCache.reset(format);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') '(' is not followed by whitespace. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.whitespace.ParenPadCheck)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') ')' is not preceded with whitespace. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.whitespace.ParenPadCheck)